lib/env-bundler.js

```diff
@@ -715,30 +721,44 @@ const defaultComponentDescriptorProcessor = function(type, location, code, compo
     });
 }
 
-const assetComponentDescriptorProcessor = function(type, location, code, components) {
+const assetComponentDescriptorProcessor = async function(type, location, code, components) {
   const typePlural = typeToPlural(type);
   const codeExtractor = componentCodeExtractors[type];
 
+  const token = await getToken();
```

Comment:
You shouldn't need to create a new token again.
Please reuse the main token that was generated at the beginning of the extraction.

Reply:
The main token is only in the scope of `getComponents()`, which is part of the `inquireComponents()` call stack; while `assetComponentDescriptorProcessor()` is part of `generateBundle()`.
I could move the token to `interactiveSession()` and just pass it into `inquireComponents()` and `generateBundle()`, which means the token gets generated once the session starts. What do you think?

Reply:
That sounds perfect.

Reply:
I instead moved the token to the topmost scope of `env-bundler.js` since some non-function definitions (e.g. `collectQuestions`) require the token within their scope (e.g. a function property calling `getComponents()`), but it's not possible to pass the token directly to them since they aren't functions and are also declared in the topmost scope.